delete-old-branches/delete_old_branches.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python3
+
+import os
+import json
+import typing
+import urllib.error
+import urllib.parse
+import urllib.request
+import re
+import email
+import subprocess
+
+
+#
+# Load settings
+#
+class Settings:
+    def __init__(self):
+        self.base_uri=os.environ["GITHUB_API_URL"]
+        self.workspace=os.environ["GITHUB_WORKSPACE"]
+        self.repository=os.environ["GITHUB_REPOSITORY"]
+
+        self.repo_token=os.environ["INPUT_REPO_TOKEN"]
+        self.date=os.environ["INPUT_DATE"]
+        self.dry_run=os.environ.get("INPUT_DRY_RUN", "true") == "true"
+
+        self.branch_regex=os.environ.get("INPUT_BRANCH_REGEX", "")
+
+        self.excluded_branches = os.environ.get("INPUT_EXCLUDED_BRANCHES", "main,master,develop").split(",")
+        self.excluded_branch_regex=os.environ.get("INPUT_EXCLUDED_BRANCH_REGEX", "")
+        self.exclude_open_pr_branches=os.environ.get("INPUT_EXCLUDE_OPEN_PR_BRANCHES", "true") == "true"
+
+
+SETTINGS = Settings()
+
+#
+# GitHub API client
+#
+class Response(typing.NamedTuple):
+    body: str
+    headers: email.message
+    status: int
+
+    def json(self) -> typing.Any:
+        """
+        Decode body's JSON.
+
+        Returns:
+            Pythonic representation of the JSON object
+        """
+        return json.loads(self.body)
+
+def request(
+    url: str,
+    method: str = "GET",
+) -> Response:
+    headers = {"Accept": "application/json", "Authorization": f"Bearer {SETTINGS.repo_token}"}
+
+    httprequest = urllib.request.Request(
+        url, headers=headers, method=method
+    )
+
+    try:
+        with urllib.request.urlopen(httprequest) as httpresponse:
+            response = Response(
+                headers=httpresponse.headers,
+                status=httpresponse.status,
+                body=httpresponse.read().decode(
+                    httpresponse.headers.get_content_charset("utf-8")
+                ),
+            )
+    except urllib.error.HTTPError as e:
+        response = Response(
+            body=str(e.reason),
+            headers=e.headers,
+            status=e.code,
+        )
+
+    return response
+
+#
+# Branch checks
+#
+def _excluded_branch(branch: str) -> bool:
+    """
+    Return True if the branch is considered a default branch
```

Comment:
*nit* this is checking if the branch is in excluded_branches, right, not a default branch a the repo? Should we update the description and variable name to reflect that?

Reply:
Good call TY